### PR TITLE
Add more detailed errors to clarify hash issues

### PR DESCRIPTION
**Pull request by Draylar**
This PR adds tweaked error messages for when hashes don't match up. 

~~I also noticed some hashes come back (might be from CF or Discord?) as uppercase when hash-providing tools claim they are lowercase. I added a special warning for it.~~ Might be Powershell doing the uppercasing, just going to remove it in favor of keeping all errors the same here.

image

### CODE REVIEW

**Comment by Draylar** [comment]
Scratch that, just going to remove the special handling for uppercasing. One second.

**Comment by comp500** [comment]
Uppercase hex haha yes

File: core/mod.go [hunk 1]
```diff
@@ -146,8 +147,11 @@ func (m Mod) DownloadFile(dest io.Writer) error {
 	}
 
 	calculatedHash := hex.EncodeToString(h.Sum(nil))
+
+	// Check if the hash of the downloaded file matches the expected hash.
 	if calculatedHash != m.Download.Hash {
-		return errors.New("hash of saved file is invalid")
+		return errors.New(fmt.Sprintf("Hash of downloaded file does not match with expected hash!\n download hash: %s\n expected hash: %s\n", calculatedHash, m.Download.Hash))
```
Comment (comp500) [inline core/mod.go:153]:
```suggestion
		return fmt.Errorf("Hash of downloaded file does not match with expected hash!\n download hash: %s\n expected hash: %s\n", calculatedHash, m.Download.Hash)
```
Just a bit cleaner :)

**Review by comp500 — CHANGES_REQUESTED** [review]
Looks good (although I should probably pass this data up the chain a bit, I'll probably clean it up in the Rust port), could just be cleaned up a bit.

**Review by comp500 — APPROVED** [review]

**Pull request merged**